backend/app/core/utils.py

```
@@ -23,18 +32,41 @@ def calculate_total_hours(clock_in: datetime, clock_out: Optional[datetime] = No
-        raise FileUploadError(detail=f"File size exceeds limit of {max_size / (1024 * 1024)} MB")
+    
+    # Check file size by reading stream
+    try:
+        content = await file.read()
+        file_size = len(content)
+        if file_size > max_size:
+            raise FileUploadError(detail=f"File size exceeds limit of {max_size / (1024 * 1024)} MB")
+        # Reset file pointer to start
+        await file.seek(0)
+    except Exception as e:
+        raise FileUploadError(detail=f"Failed to validate file size: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)